app/models/repaymentHistory/RepaymentHistoryUtils.scala

```diff
@@ -43,9 +43,41 @@ object RepaymentHistoryUtils {
     }
   }
 
-  def combinePaymentHistoryData(payments: List[Payment], repayments: List[RepaymentHistory], isAgent: Boolean,
-                               MFACreditsEnabled: Boolean, CutOverCreditsEnabled: Boolean, languageUtils: LanguageUtils
-                               )(implicit messages: Messages): List[PaymentHistoryEntry] = {
+  private def getRefundsLink(isAgent: Boolean) = {
+    if (isAgent) {
+      "agents/refund-to-taxpayer"
+    } else {
+      "refund-to-taxpayer"
+    }
+  }
+
+  private def groupedPayments(payments: List[PaymentHistoryEntry]): List[(Int, List[PaymentHistoryEntry])] = {
+    def sortPayments(payments: List[PaymentHistoryEntry]) = {
+      payments
+        .map(payment => (LocalDate.parse(payment.date).toEpochDay, payment))
+        .sortWith((left, right) => left._1 < right._1)
+        .map { case (_, payments) => payments }
+    }
+
+    payments.groupBy[Int] { payment => {
+      LocalDate.parse(payment.date).getYear
+    }
+    }.toList.sortBy(_._1).reverse
+      .map { case (year, payments) => (year, sortPayments(payments)) }
+  }
+
```

Comment:
I guess this grouping would make more sense on the model level or as a helper function / utils i.e?